cmd/drone-docker/main.go

```diff
@@ -290,5 +293,13 @@ func run(c *cli.Context) error {
 		}
 	}
 
+	// Make sure we always have a name when commit.sha is not available.
+	// See https://github.com/drone-plugins/drone-docker/issues/229
+	if plugin.Build.Name == "" {
+		// Use hex encoding as docker requires lowercase.
+		// Note: we can't use 64 characters as it conflicts with sha256 image hashes.
+		plugin.Build.Name = uniuri.NewLenChars(40, hexChars)
```

Comment:
40 chars? Why do you want to define such a long random name?

Reply:
40 matches the keyspace of git's sha1 so it has similar properties in terms of collisions.